CMakeLists.txt

```
@@ -50,21 +50,49 @@ endif()
 
 # Common setup
 
-if(NOT DEFINED ${CMAKE_BUILD_TYPE})
-  set(CMAKE_BUILD_TYPE "Debug")
-endif()
-
 find_package(Threads REQUIRED)
 add_subdirectory(Utilities)
 
 # DPC++ setup
 
 if (SYCL_ACADEMY_USE_DPCPP)
-# For DPCPP module to work, find_package to work, use the options
-# -DCMAKE_C_COMPILER=icx -DCMAKE_CXX_COMPILER=icpx on the command line.
-# See /opt/intel/oneapi/compiler/latest/linux/IntelSYCL/ReadMeSYCL.txt
-  set(IntelSYCL_DIR /opt/intel/oneapi/compiler/latest/linux/IntelSYCL)
-  find_package(IntelSYCL REQUIRED)
+  if (DEFINED SYCL_TRIPLE)
+    list(APPEND CMAKE_CXX_FLAGS "-fsycl-targets=${SYCL_TRIPLE}")
+  endif()
+  if (DEFINED SYCL_ARCH)
+    if (${SYCL_ARCH} MATCHES pvc)
+      list(APPEND CMAKE_CXX_FLAGS "-Xs" "\"-device pvc\"")
```

I would do:
```suggestion
    if (${SYCL_TRIPLE} MATCHES spir64_gen)
      list(APPEND CMAKE_CXX_FLAGS "-Xs" "-device ${SYCL_ARCH}")
```
to allow other Intel GPUs as well (e.g. Arc or even iGPUs).

However, testing this I actually can't get either solution to work with oneAPI release FindIntelSYCL. They do a feature test where they compile a basic file with the provided flags, but somehow all quotes are removed on the way. This ends up compiling with `-Xs -device pvc`, i.e. the "pvc" argument is treated as separate and it fails with `icpx: error: no such file or directory: 'pvc'`. I tried a few combinations of quote escaping here, but didn't manage to make it work.
`ocloc` doesn't accept `-device=pvc` syntax, it requires a space.

If you don't have further ideas on how to fix that, maybe we can just drop the `spir64_gen` support for now. I imagine the generic `spir64` is still fine for the examples in this repo and we wouldn't gain much from device-specific features.